### DISCUSSION TITLE
quick/scaffold-api-run-and-test

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#75] [General]
# quick/scaffold-api-run-and-test

通过命令行启动服务，并使用swagger生成接口文档。支持创建用户、查询用户信息、修改用户数据以及删除用户的RESTful API接口。同时支持使用curl命令进行接口测试，提供详细的校验规则和错误码以确保数据的准确性和可靠性。

https://goframe.org/quick/scaffold-api-run-and-test

**Comment by Grace-Cao** [comment]
提示：创建和修改接口的方式颠倒了。

**Reply by gqcn** [reply]
你好，我们修正将`post`作为`Create`的方式，`put`作为`Update`的方式。

**Comment by YRhappyer** [comment]
您好，我是一个小白，刚刚接触，在最后测试创建的时候，出现了{"code":51,"message":"The Name field is required","data":null}，网上也没搜到解决办法，为什么呢，怎么解决呀

**Reply by YRhappyer** [reply]
在终端是这样，在postman可以正常运行

**Reply by lxb-foxer** [reply]
加一，我也是这样的问题，是什么原因造成的呢

**Comment by wangyunan2018** [comment]
您好，我按照文档一步一步做的，在启动项目的时候报错了，报错如下：
PS D:\code\golang\golang-test\goframe-demo1\demo> gf run .\main.go
build: .\main.go
go build -o .\main.exe  .\main.go
build error: 
# github.com/gogf/gf/v2/net/ghttp
C:\Users\17822\go\pkg\mod\github.com\gogf\gf\v2@v2.9.0\net\ghttp\ghttp_server.go:300:9: table.SetHeader undefined (type *tablewriter.Table has no field or method SetHeader)
C:\Users\17822\go\pkg\mod\github.com\gogf\gf\v2@v2.9.0\net\ghttp\ghttp_server.go:301:9: table.SetRowLine undefined (type *tablewriter.Table has no field or method SetRowLine)
C:\Users\17822\go\pkg\mod\github.com\gogf\gf\v2@v2.9.0\net\ghttp\ghttp_server.go:302:9: table.SetBorder undefined (type *tablewriter.Table has no field or method SetBorder)
C:\Users\17822\go\pkg\mod\github.com\gogf\gf\v2@v2.9.0\net\ghttp\ghttp_server.go:303:9: table.SetCenterSeparator undefined (type *tablewriter.Table has no field or method SetCenterSeparator)
exit status 1

找到报错的go文件，这几行在报错：
if len(routes) > 0 {
		buffer := bytes.NewBuffer(nil)
		table := tablewriter.NewWriter(buffer)
		table.SetHeader(headers)
		table.SetRowLine(true)
		table.SetBorder(false)
		table.SetCenterSeparator("|")

go的版本是1.23.3，启动项目的时候都能正常运行，不知道问题出现在哪儿，报了跟业务代码不相干的错，其中，mysql数据库密码是强密码，密码中带了@符号，和后面的@符号是否有冲突呢？现在项目运行不起来，不知道对不对

**Reply by wangyunan2018** [reply]
尝试了go get -u github.com/olekukonko/tablewriter, go mod，清除go 缓存都没用，后面我把有问题的那个ghttp_server.go中的header相关的注释掉了，然后就正常启动了，且能正常进行CURD，有点儿奇怪，起初创建好项目的时候都能正常运行，写完user接口，然后注册路由，启动服务就报错了
func (s *Server) doRouterMapDump() {
	if !s.config.DumpRouterMap {
		return
	}

	var (
		ctx                          = context.TODO()
		routes                       = s.GetRoutes()
		isJustDefaultServerAndDomain = true
		// headers                      = []string{
		// 	"SERVER", "DOMAIN", "ADDRESS", "METHOD", "ROUTE", "HANDLER", "MIDDLEWARE",
		// }
	)
	for _, item := range routes {
		if item.Server != DefaultServerName || item.Domain != DefaultDomainName {
			isJustDefaultServerAndDomain = false
			break
		}
	}
	// if isJustDefaultServerAndDomain {
	// 	headers = []string{"ADDRESS", "METHOD", "ROUTE", "HANDLER", "MIDDLEWARE"}
	// }
	if len(routes) > 0 {
		buffer := bytes.NewBuffer(nil)
		table := tablewriter.NewWriter(buffer)
		// table.SetHeader(headers)
		// table.SetRowLine(true)
		// table.SetBorder(false)
		// table.SetCenterSeparator("|")

**Comment by johnson0205** [comment]
启动后这些接口打印怎么能关掉

**Reply by johnson0205** [reply]
找到了config里面配置server.dumpRouterMap: true
